### ISSUE TITLE
Discrepancies between GPL 1.0 & 2.0 canonical texts and their associated SPDX templates

### ISSUE THREAD

**Issue by pmonks** [spdx/license-list-XML#2568]
There are discrepancies between FSF's canonical GPL-1.0 and GPL-2.0 texts and their associated SPDX templates that cause matching to fail in downstream software that performs matching.

Specifically:

* The [canonical `GPL-1.0` text](https://www.gnu.org/licenses/old-licenses/gpl-1.0.txt) no longer includes a physical address on line 6, and has added a URL in that location instead.  Neither of these changes are taken into account in either the [`GPL-1.0-only`](https://github.com/spdx/license-list-XML/blob/main/src/GPL-1.0-only.xml#L41-L44) or [`GPL-1.0-or-later`](https://github.com/spdx/license-list-XML/blob/main/src/GPL-1.0-or-later.xml#L23-L26) SPDX templates.
* Similarly, the [canonical `GPL-2.0` text](https://www.gnu.org/licenses/old-licenses/gpl-2.0.txt) now has a URL on line 6.  While the [`GPL-2.0-only`](https://github.com/spdx/license-list-XML/blob/main/src/GPL-2.0-only.xml#L44-L48) and [`GPL-2.0-or-later`](https://github.com/spdx/license-list-XML/blob/main/src/GPL-2.0-or-later.xml#L24-L28) correctly handle the (now optional) physical address, neither of them handle the (presumably) optional URL that is now included in the canonical text.

Note: if the SPDX project has contacts over at the FSF it may be worth asking them if it might be possible to notify the SPDX project whenever they make changes of any kind to their license texts (even/especially "legally irrelevant" ones).  Previous issues (including #2204, #1995, #1973, #1972) suggest that the FSF are quite liberal about making such changes and thereby inadvertently breaking SPDX license matching randomly.

**Comment by pmonks** [comment]
It appears the same issue exists in the (old) LGPL variants too:

* [canonical `LGPL-2.0` text](https://www.gnu.org/licenses/old-licenses/lgpl-2.0.txt) vs [`LGPL-2.0-only` SPDX template](https://github.com/spdx/license-list-XML/blob/main/src/LGPL-2.0-only.xml#L40-L45) and [`LGPL-2.0-or-later` SPDX template](https://github.com/spdx/license-list-XML/blob/main/src/LGPL-2.0-or-later.xml#L26-L31)
* [canonical `LGPL-2.1` text](https://www.gnu.org/licenses/old-licenses/lgpl-2.1.txt) vs [`LGPL-2.1-only` SPDX template](https://github.com/spdx/license-list-XML/blob/main/src/LGPL-2.1-only.xml#L44-L47) and [`LGPL-2.1-or-later` SPDX template](https://github.com/spdx/license-list-XML/blob/main/src/LGPL-2.1-or-later.xml#L26-L29)

The `LGPL-3.0-*` SPDX templates does appear to be aligned with the FSF's [canonical `LGPL-3.0` text](https://www.gnu.org/licenses/lgpl-3.0.txt), however.

**Comment by jlovejoy** [comment]
good catch! And since the copyright notice in this (somewhat rare case) is on the license itself, this is not a situation where for matching purposes it might be ignored as part of the copyright notice.

the good news is that this can easily be accommodated with the alt tag. 

@pmonks - do you want to prepare a PR?

**Comment by szepeviktor** [comment]
Is the change in the sample copyright disclaimer relevant here?

```diff
-  <signature of Ty Coon>, 1 April 1989
-  Ty Coon, President of Vice
+  <signature of Moe Ghoul>, 1 April 1989
+  Moe Ghoul, President of Vice
```

https://www.gnu.org/licenses/old-licenses/gpl-2.0.txt

**Comment by jlovejoy** [comment]
this just needs a PR to address these variations

**Comment by Pizza-Ria** [comment]
+1 to merging the address vs URL change. Perhaps "The GNU Volunteer Coordinators [<gvc@gnu.org>](mailto:gvc@gnu.org) can assist you if you would like to help developing GNU software." would be a decent place to pose your query on notice re breaking changes.

**Comment by swinslow** [comment]
As discussed in #2632, there appears to be a technical issue with the XML markup for this -- or possibly just a misunderstanding on my part.  :)  

In any case, I think we may need @goneall's input to resolve fully, so I'm moving this to 3.27.0.

**Comment by pmonks** [comment]
I don't know what the next steps are for this issue - perhaps it could be reassigned to someone who does (or unassigned if that's not clear)?  I'm unable to remove myself as the assigneee.

**Comment by goneall** [comment]
@pmonks - I just added a comment to the PR - I think this is now fixed, we just need to re-run the CI to be sure

**Comment by pmonks** [comment]
> IMHO there is no such thing as a single canonical license text.

The definition of "canonical" I'm using here is "the current license text published by the license author". If the SPDX community has previously adopted some other label for this important concept, I'd be keen to learn what it is.

> The actual history of the GNU license texts shows that whoever manage them do not seem to care much to announce any small changes and do not seem to use any version control or basic public review process. 

It certainly seems that way to me too, but looking at it charitably I'm open to the possibility that the FSF is simply unaware of the disruption their current processes cause downstream, and that they'd be open to feedback about that from an affected party (such as the SPDX project).

**Comment by johnsu01** [comment]
> [@pmonks](https://github.com/pmonks) re:
> 
> > Previous issues (including [#2430](https://github.com/spdx/license-list-XML/issues/2430), [#2204](https://github.com/spdx/license-list-XML/issues/2204), [#1995](https://github.com/spdx/license-list-XML/issues/1995), [#1973](https://github.com/spdx/license-list-XML/issues/1973), [#1972](https://github.com/spdx/license-list-XML/issues/1972)) suggest that the FSF are quite liberal about making such changes and thereby inadvertently breaking SPDX license matching randomly.
> 
> The actual history of the GNU license texts shows that whoever manage them do not seem to care much to announce any small changes and do not seem to use any version control or basic public review process.

FWIW, the version control is public, ie https://web.cvs.savannah.gnu.org/viewvc/www/www/licenses/old-licenses/gpl-2.0-body.include?sortby=date&view=log

**Comment by jlovejoy** [comment]
closed by #2954

**Comment by pombredanne** [comment]
> FWIW, the version control is public, ie https://web.cvs.savannah.gnu.org/viewvc/www/www/licenses/old-licenses/gpl-2.0-body.include?sortby=date&view=log

@johnsu01 Thank you ... I was not aware of that resource. But the link seems down and is timing out. Do you have an updated one by chance?

**Comment by johnsu01** [comment]
Philippe Ombredanne ***@***.***> writes:

> pombredanne left a comment (spdx/license-list-XML#2568)
>
>> FWIW, the version control is public, ie https://web.cvs.savannah.gnu.org/viewvc/www/www/licenses/old-licenses/gpl-2.0-body.include?sortby=date&view=log
>
> @johnsu01 Thank you ... I was not aware of that resource. But the link seems down and is timing out. Do you have an updated one by chance?

I believe the downtime is temporary (but frequent) and that this is
still the correct link. One could also just check out the CVS repo
(still needs the server's cooperation, but maybe a different part of the
server..)

-john
